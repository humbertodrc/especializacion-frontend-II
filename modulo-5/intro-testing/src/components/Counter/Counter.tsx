import React from "react";
import {useState} from "react";
import reactLogo from "../../assets/react.svg";

export interface CounterInterface {
	value: number;
}

const Counter: React.FC<CounterInterface> = ({value}) => {
	const [count, setCount] = useState(value);
	return (
		<div className="App">
			<h1>Testing React</h1>
			<h2>{count}</h2>
			<div className="card">
				<button onClick={() => setCount((count) => count + 1)}>+1</button>
				<button onClick={() => setCount((count) => count - 1)}>-1</button>
			</div>
		</div>
	);
};

export default Counter;
