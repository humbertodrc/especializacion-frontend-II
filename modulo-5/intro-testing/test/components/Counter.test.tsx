import React from 'react';
import {render, screen, fireEvent} from "@testing-library/react"
import Counter from '../../src/components/Counter/Counter';
import '@testing-library/jest-dom';

describe('Pruebas en <Counter />', () => {

  // Arrange
  const value = 50;

  test('Debe conincidir con el snapshot', () => {
    // Act
    const { container } = render(<Counter value={value} />)
    
    // screen.debug()

    // Assert
    expect(container).toMatchSnapshot()
  })

  test('Se debe mostar el valor inicial de 50', () => {
    // Act
    render(<Counter value={value} />)

    // Assert 
    // Para que funcione el toBeInTheDocument se debe importarimport '@testing-library/jest-dom';
    expect(screen.getByText('50')).toBeInTheDocument()
    expect(screen.getByText('50')).toHaveTextContent('50')
    expect(screen.getByText(/50/)).toBeInTheDocument()
    expect(screen.getByText(/50/)).toBeTruthy()

  })

  test('Debe incrementar el valor en +1', () => {
    // Act
    render(<Counter value={value} />)
    const btnIncrement = screen.getByText('+1')

    fireEvent.click(btnIncrement)

    // screen.debug()

    // Assert
    expect(screen.getByText('51')).toBeInTheDocument()
  })

  test('Debe decrementar el valor en -1', () => {
    // Act
    render(<Counter value={value} />)
    const btnDecrement = screen.getByText('-1')

    fireEvent.click(btnDecrement)

    // screen.debug()

    // Assert
    expect(screen.getByText('49')).toBeInTheDocument()
  })

})