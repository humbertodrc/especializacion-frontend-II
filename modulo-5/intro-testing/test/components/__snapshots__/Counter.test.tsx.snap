// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Counter /> Debe conincidir con el snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Testing React
    </h1>
    <h2>
      50
    </h2>
    <div
      class="card"
    >
      <button>
        +1
      </button>
      <button>
        -1
      </button>
    </div>
  </div>
</div>
`;
